aten/src/ATen/native/AdaptiveAveragePooling3d.cpp

```diff
@@ -215,22 +250,46 @@ static void adaptive_avg_pool3d_backward_out_frame(
   }
 }
 
+template <typename scalar_t>
+static void adaptive_avg_pool3d_backward_out_frame(
+    scalar_t *gradInput_p,
+    scalar_t *gradOutput_p,
+    int64_t sizeB,
+    int64_t sizeD,
+    int64_t isizeT,
+    int64_t isizeH,
+    int64_t isizeW,
+    int64_t osizeT,
+    int64_t osizeH,
+    int64_t osizeW)
+{
+  int64_t b;
+#pragma omp parallel for private(b)
+  for (b = 0; b < sizeB; b++)
+  {
+    adaptive_avg_pool3d_backward_single_out_frame(
+      gradInput_p+b*sizeD*isizeT*isizeH*isizeW,
+      gradOutput_p+b*sizeD*osizeT*osizeH*osizeW,
+      sizeD,
+      isizeT, isizeH, isizeW, osizeT, osizeH, osizeW);
+  }
+}
+
 Tensor& adaptive_avg_pool3d_backward_out_cpu_template(
     Tensor& gradInput,
     const Tensor& gradOutput_,
     const Tensor& input) {
-  /* sizes */
-  int sizeD = input.size(-4);
-  int isizeT = input.size(-3);
-  int isizeH = input.size(-2);
-  int isizeW = input.size(-1);
-  int osizeT = gradOutput_.size(-3);
-  int osizeH = gradOutput_.size(-2);
-  int osizeW = gradOutput_.size(-1);
-
   /* get contiguous gradOutput */
   auto gradOutput = gradOutput_.contiguous();
 
+  /* sizes */
+  int64_t sizeD = input.size(-4);
+  int64_t isizeT = input.size(-3);
+  int64_t isizeH = input.size(-2);
+  int64_t isizeW = input.size(-1);
+  int64_t osizeT = gradOutput_.size(-3);
+  int64_t osizeH = gradOutput_.size(-2);
+  int64_t osizeW = gradOutput_.size(-1);
```

Comment:
Again, extreme nitpick, but it still calls `gradOutput_.size(-3)` and not `gradOutput.size(-3)`. :)

Reply:
Thanks.